neurodamus/ngv.py

```diff
@@ -19,19 +19,136 @@
 from .utils.pyutils import append_recarray, bin_search
 
 
+class GlutList:
+    """GlutList is a list-like container that combines a standard
+    list (the body) with a separate tail element.
+
+    It behaves like a normal list with one extra element at the end,
+    enabling efficient operations without shifting the tail.
+    Supports indexing (including negative), iteration, equality
+    comparison with lists, and conversion to a standard list.
+    """
+
+    __hash__ = None
+
+    def __init__(self, body=None, tail=None):
+        if body is None:
+            body = []
+        self._body = list(body)
+        self._tail = tail
+
+    def __len__(self):
+        return len(self._body) + 1
+
+    def _sanitize_index(self, index):
+        total_len = len(self)
+        og_index = index
+        if index < 0:
+            index += total_len
+        if index < 0 or index >= total_len:
+            raise IndexError(f"GlutList index {og_index} out of range")
+        return index
+
+    def __getitem__(self, index):
+        index = self._sanitize_index(index)
+        return self._body[index] if index != len(self) - 1 else self._tail
+
+    def append(self, item):
+        self._body.append(item)
+
+    def pop(self, index=-2):
+        index = self._sanitize_index(index)
+        if index == len(self) - 1:
+            raise IndexError("GlutList tail cannot be popped")
+        self._body.pop(index)
+
+    def __setitem__(self, index, value):
+        index = self._sanitize_index(index)
+        if index == len(self) - 1:
+            self._tail = value
+        else:
+            self._body.__setitem__(index, value)
+
+    def __iter__(self):
+        yield from self._body
+        yield self._tail
+
+    def to_list(self):
+        return [*self._body, self._tail]
+
+    def __eq__(self, other):
+        if isinstance(other, GlutList):
+            return self._body == other._body and self._tail == other._tail
+        if isinstance(other, list):
+            return self.to_list() == other
+        return NotImplemented
+
+    @property
+    def tail(self):
+        return self._tail
+
+    @tail.setter
+    def tail(self, value):
+        self._tail = value
+
+    def __str__(self):
+        return str(self.to_list())
+
+
 class Astrocyte(BaseCell):
-    __slots__ = ("_glut_list", "_nseg_warning", "_secidx2names")
+    __slots__ = ("_glut_list", "_nseg_warning", "section_names")
 
     def __init__(self, gid, meinfos, circuit_conf):
-        """Instantiate a new Cell from node info."""
+        """Initialize an Astrocyte cell.
+
+        - Load and attach morphology to the cell.
```

Comment:
not sure this is needed here, since the comments say the same thing

Reply:
yes, redundant. Removed